source/common/http/http1/codec_impl.cc

```diff
@@ -562,16 +562,6 @@ int ConnectionImpl::onHeadersCompleteBase() {
       ENVOY_CONN_LOG(trace, "codec entering upgrade mode.", connection_);
```

Comment:
Gotcha, perhaps the PR author assumed full connection sanitization was happening in the h2c case, but it only removed upgrade/http2settings from connection header. Speaking of, can you make the quick change from
`current_headers.remove(Headers::get().Http2Settings);` 
`current_headers.removeHttp2Settings();`

Reply:
Http2Settings is not an O(1) header currently?

Reply:
Oh, maybe it's not that significant, you just don't have to do the lookup at all if you use `remove##name`.